gdnative-core/src/macros.rs

```diff
@@ -88,7 +85,8 @@ macro_rules! godot_gdnative_terminate {
                 return;
             }
 
-            let __result = ::std::panic::catch_unwind(|| $callback(options));
+            let term_info = $crate::TerminateInfo::new(options);
```

Comment:
I think the constructor of `TerminateInfo` may panic? Might be better to put it in the `catch_unwind` closure.

Reply:
Updated! Good catch.